chaoslib/litmus/aws-ssm-chaos/lib/ssm/aws-ssm-chaos-by-id.go

```
@@ -0,0 +1,84 @@
+	"github.com/litmuschaos/litmus-go/pkg/types"
+	"github.com/litmuschaos/litmus-go/pkg/utils/common"
+	"github.com/pkg/errors"
+)
+
+var (
+	err           error
+	inject, abort chan os.Signal
+)
+
+//PrepareAWSSSMChaosByID contains the prepration and injection steps for the experiment
+func PrepareAWSSSMChaosByID(experimentsDetails *experimentTypes.ExperimentDetails, clients clients.ClientSets, resultDetails *types.ResultDetails, eventsDetails *types.EventDetails, chaosDetails *types.ChaosDetails) error {
+
+	// inject channel is used to transmit signal notifications.
+	inject = make(chan os.Signal, 1)
+	// Catch and relay certain signal(s) to inject channel.
+	signal.Notify(inject, os.Interrupt, syscall.SIGTERM)
+
+	// abort channel is used to transmit signal notifications.
+	abort = make(chan os.Signal, 1)
+	// Catch and relay certain signal(s) to abort channel.
+	signal.Notify(abort, os.Interrupt, syscall.SIGTERM)
+
+	//Waiting for the ramp time before chaos injection
+	if experimentsDetails.RampTime != 0 {
+		log.Infof("[Ramp]: Waiting for the %vs ramp time before injecting chaos", experimentsDetails.RampTime)
+		common.WaitForDuration(experimentsDetails.RampTime)
+	}
+
+	//create and upload the ssm document on the given aws service monitoring docs
+	if err = ssm.CreateAndUploadDocument(experimentsDetails.DocumentName, experimentsDetails.DocumentType, experimentsDetails.DocumentFormat, experimentsDetails.DocumentPath, experimentsDetails.Region); err != nil {
+		return errors.Errorf("fail to create and upload ssm doc, err: %v", err)
+	}
+	experimentsDetails.IsDocsUploaded = true
+	log.Info("[Info]: SSM docs uploaded successfully")
+
+	// watching for the abort signal and revert the chaos
+	go lib.AbortWatcher(experimentsDetails, abort)
+
+	//get the instance id or list of instance ids
+	instanceIDList := strings.Split(experimentsDetails.EC2InstanceID, ",")
+	if len(instanceIDList) == 0 {
+		return errors.Errorf("no instance id found to terminate")
```

we can return the appropriate error